lib/shelley/test/unit/Cardano/Wallet/Shelley/TransactionSpec.hs

```diff
@@ -2458,6 +2473,20 @@ instance Arbitrary FeePolicy where
         LinearFee . uncurry LinearFunction
             <$> shrink (intercept, slope)
 
+newtype TxBalanceSurplus a = TxBalanceSurplus {unTxBalanceSurplus :: a}
+    deriving (Eq, Show)
+
+instance Arbitrary (TxBalanceSurplus Coin) where
+    -- We want to test cases where the surplus is zero. So it's important that
+    -- we do not restrict ourselves to positive coins here.
+    arbitrary = TxBalanceSurplus <$> frequency
```

Comment:
Ah! 👍

Reply:
To be honest, I think the global `Arbitrary Coin` instance probably should include `Coin 0`.

I tried this, and all the tests in the module _seemed to pass_. But still out of caution retraining from changing this now. It would be easier to change after we reorganize the module structure.